### PR TITLE
dev-vcs/git-cola: bump unstable 2.6, stable 2.4, drop old

### PR DESCRIPTION
**Pull request by lkraav**
2.4 is from nearly 9 months ago already, no open bugs, looks safe to bump.

### CODE REVIEW

**Comment by monsieurp** [comment]
@mrueg @dev-zero

File: dev-vcs/git-cola/git-cola-2.4.ebuild [hunk 1]
```diff
@@ -15,7 +15,7 @@ SRC_URI="https://github.com/${PN}/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+KEYWORDS="amd64 x86 amd64-linux x86-linux"
```
Comment (Coacher) [inline dev-vcs/git-cola/git-cola-2.4.ebuild]: Stabilization procedure in Gentoo doesn't work like that.

File: dev-vcs/git-cola/git-cola-2.6.ebuild [hunk 2]
```diff
@@ -7,7 +7,7 @@ EAPI=5
 PYTHON_COMPAT=( python2_7 python3_{3,4} )
 DISTUTILS_SINGLE_IMPL=true
 
-inherit distutils-r1 readme.gentoo virtualx
+inherit distutils-r1 readme.gentoo-r1 virtualx
```
Comment (Coacher) [inline dev-vcs/git-cola/git-cola-2.6.ebuild]: readme.gentoo-r1 requires explicit pkg_postinst. Thus transition to readme.gentoo-r1 is incomplete.

File: dev-vcs/git-cola/git-cola-2.6.ebuild [hunk 3]
```diff
@@ -32,9 +32,7 @@ DEPEND="${RDEPEND}
 	test? ( dev-python/nose[${PYTHON_USEDEP}] )
 "
 
-PATCHES=(
-	"${FILESDIR}"/${PN}-2.2.1-disable-live-tests.patch
-)
+PATCHES=()
```
Comment (Coacher) [inline dev-vcs/git-cola/git-cola-2.6.ebuild]: Empty PATCHES should be removed.

File: dev-vcs/git-cola/git-cola-2.6.ebuild [hunk 4]
```diff
@@ -32,9 +32,7 @@ DEPEND="${RDEPEND}
 	test? ( dev-python/nose[${PYTHON_USEDEP}] )
```
Comment (Coacher) [inline dev-vcs/git-cola/git-cola-2.6.ebuild]: Clearly ebuild wasn't tested properly because testsuite requires dev-python/mock as well.

**Comment by lkraav** [comment]
Thanks for the feedback. Everything you say is correct. Unfortunately I can't find time to further patch the ebuild up until week after next. Perhaps this has still been useful to get someone else finishing this off.

File: dev-vcs/git-cola/git-cola-2.6.ebuild [hunk 5]
```diff
@@ -87,7 +85,7 @@ python_install_all() {
 		prefix="${EPREFIX}/usr" \
 		install
 
-	python_fix_shebang "${ED}/usr/share/git-cola/bin/git-xbase"
+	python_fix_shebang "${ED}/usr/share/git-cola/bin/git-xbase" "${ED}"/usr/bin/git-cola
```
Comment (Coacher) [inline dev-vcs/git-cola/git-cola-2.6.ebuild]: Unneeded change.
See https://docs.python.org/2/distutils/setupscript.html#installing-scripts
and https://github.com/git-cola/git-cola/blob/v2.6/setup.py#L35

**Comment by idella** [comment]
ikraav if you are serious about this you will remake a bug of this and assign and CC accordingly.

**Comment by lkraav** [comment]
I will give it a shot, but timeline suffers. Just got a fresh newborn.

**Comment by idella** [comment]
` Just got a fresh newborn.`   OOOOh

**Comment by mgorny** [comment]
@lkraav, a bit late gratz! ;-)

@gentoo/proxy-maint, could one of us please finish this one and file the stablereq? Shouldn't take long.

**Comment by gktrk** [comment]
Doesn't look like the problems pointed out by @Coacher have been addressed yet.

@lkraav ping

Do we have the permission of the maintainers to add a user to the metadata?

**Comment by Coacher** [comment]
Here's a working ebuild where problems pointed above and several others are fixed:
https://github.com/Coacher/bonespirit/blob/master/dev-vcs/git-cola/git-cola-9999.ebuild

I'm only interested in maintaining it in my overlay, but feel free to pick useful fixes if you find any.

**Comment by lkraav** [comment]
`2.7` was just released https://github.com/git-cola/git-cola/blob/v2.7/share/doc/git-cola/relnotes/v2.7.rst

> git cola now depends on QtPy and includes a bundled copy of the qtpy library. If you are packaging git cola and would prefer to use qtpy from your distribution instead of the built-in version then use make NO_VENDOR_LIBS=1 when building git cola. This will prevent vendored libraries from being installed.

**Comment by monsieurp** [comment]
@lkraav hi. what's the deal with this PR? can we get on with it? I'm going to close it since problems noted by @Coacher haven't been addressed yet. feel free to reopen it.

**Comment by lkraav** [comment]
No time to investigate. It seemed like somebody more knowledgeable with the internals here was going to help out, but looks like that never materialized. I'll pick it up again when time permits.

**Pull request closed without merging**